pandas/_libs/interval.pyx

```diff
@@ -54,7 +54,17 @@ cdef class IntervalMixin(object):
             return 0.5 * (self.left + self.right)
         except TypeError:
             # datetime safe version
-            return self.left + 0.5 * (self.right - self.left)
+            return self.left + 0.5 * self.length
+
+    @property
+    def length(self):
+        """Return the length of the Interval"""
+        try:
+            return self.right - self.left
+        except TypeError:
+            # length not defined for some types, e.g. string
+            msg = 'cannot compute length between {left} and {right}'
```

Comment:
A nitpick perhaps: IMO the form ``msg = 'cannot compute length between {left!r} and {right!r}'`` (note the ``!r``) is better, so variables get outputted as raw strings where relevant in error messages.

In general IMO variables in error messages should use ``!r``, though this is not very common on pandas.